### PR TITLE
Expose user locations on client API

### PR DESCRIPTION
**Pull request by mmadariaga**
Added user location getter on client API through terminal, friend, residential device and retail account

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Model/ResidentialDevice/ResidentialDeviceDto.php [hunk 1]
```diff
@@ -2,24 +2,102 @@
...
 class ResidentialDeviceDto extends ResidentialDeviceDtoAbstract
 {
+    const CONTEXT_STATUS = 'status';
+
+    /**
+     * @var Status[]
+     * @AttributeDefinition(
+     *     type="array",
+     *     class="Ivoz\Kam\Domain\Model\UsersLocation\Status",
+     *     description="Terminal status"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/ResidentialDevice/ResidentialDeviceDto.php]: See above

File: library/Ivoz/Provider/Domain/Model/Friend/FriendDto.php [hunk 2]
```diff
@@ -2,15 +2,92 @@
...
 class FriendDto extends FriendDtoAbstract
 {
+    const CONTEXT_STATUS = 'status';
+
+    /**
+     * @var Status[]
+     * @AttributeDefinition(
+     *     type="array",
+     *     class="Ivoz\Kam\Domain\Model\UsersLocation\Status",
+     *     description="Terminal status"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/Friend/FriendDto.php]: description="Registration status" is a description that you can use for all these endpoints

File: library/Ivoz/Provider/Domain/Model/Friend/FriendDto.php [hunk 3]
```diff
@@ -2,15 +2,92 @@
...
+     * )
+     */
+    protected $status = [];
+
+    /**
+     * @var string
+     * @AttributeDefinition(
+     *     type="string",
+     *     description="Terminal domain"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/Friend/FriendDto.php]: description="Registration domain" is a description that you can use for all these endpoints

File: library/Ivoz/Provider/Domain/Model/ResidentialDevice/ResidentialDeviceDto.php [hunk 4]
```diff
@@ -2,24 +2,102 @@
...
+     * )
+     */
+    protected $status = [];
+
+    /**
+     * @var string
+     * @AttributeDefinition(
+     *     type="string",
+     *     description="Terminal domain"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/ResidentialDevice/ResidentialDeviceDto.php]: See above

File: library/Ivoz/Provider/Application/Service/ResidentialDevice/ResidentialDeviceDtoAssembler.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     * @return ResidentialDeviceDto
+     */
+    public function toDto(EntityInterface $residentialDevice, $depth = 0, string $context = null)
+    {
+        Assertion::isInstanceOf($residentialDevice, ResidentialDeviceInterface::class);
+
+        /** @var ResidentialDeviceDto $dto */
+        $dto = $residentialDevice->toDto($depth);
+
+        if ($context !== 'status') {
```
Comment (Kaian) [inline library/Ivoz/Provider/Application/Service/ResidentialDevice/ResidentialDeviceDtoAssembler.php]: Define CONTEXT_STATUS constant like in Dtos?

File: library/Ivoz/Provider/Application/Service/RetailAccount/RetailAccountDtoAssembler.php [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+     * @return RetailAccountDto
+     */
+    public function toDto(EntityInterface $retailAccount, $depth = 0, string $context = null)
+    {
+        Assertion::isInstanceOf($retailAccount, RetailAccountInterface::class);
+
+        /** @var RetailAccountDto $dto */
+        $dto = $retailAccount->toDto($depth);
+
+        if ($context !== 'status') {
```
Comment (Kaian) [inline library/Ivoz/Provider/Application/Service/RetailAccount/RetailAccountDtoAssembler.php]: See above

File: library/Ivoz/Provider/Application/Service/Terminal/TerminalDtoAssembler.php [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+     * @return TerminalDto
+     */
+    public function toDto(EntityInterface $terminal, $depth = 0, string $context = null)
+    {
+        Assertion::isInstanceOf($terminal, TerminalInterface::class);
+
+        /** @var TerminalDto $dto */
+        $dto = $terminal->toDto($depth);
+
+        if ($context !== 'status') {
```
Comment (Kaian) [inline library/Ivoz/Provider/Application/Service/Terminal/TerminalDtoAssembler.php]: See above

File: library/Ivoz/Provider/Domain/Model/RetailAccount/RetailAccountDto.php [hunk 8]
```diff
@@ -2,15 +2,100 @@
...
 class RetailAccountDto extends RetailAccountDtoAbstract
 {
+    const CONTEXT_STATUS = 'status';
+
+    /**
+     * @var Status[]
+     * @AttributeDefinition(
+     *     type="array",
+     *     class="Ivoz\Kam\Domain\Model\UsersLocation\Status",
+     *     description="Terminal status"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/RetailAccount/RetailAccountDto.php]: See above

File: library/Ivoz/Provider/Domain/Model/RetailAccount/RetailAccountDto.php [hunk 9]
```diff
@@ -2,15 +2,100 @@
...
+     * )
+     */
+    protected $status = [];
+
+    /**
+     * @var string
+     * @AttributeDefinition(
+     *     type="string",
+     *     description="Terminal domain"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/RetailAccount/RetailAccountDto.php]: See above

File: library/Ivoz/Provider/Domain/Model/Terminal/TerminalDto.php [hunk 10]
```diff
@@ -2,14 +2,82 @@
...
 class TerminalDto extends TerminalDtoAbstract
 {
+    const CONTEXT_STATUS = 'status';
+
+    /**
+     * @var Status[]
+     * @AttributeDefinition(
+     *     type="array",
+     *     class="Ivoz\Kam\Domain\Model\UsersLocation\Status",
+     *     description="Terminal status"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/Terminal/TerminalDto.php]: See above

File: library/Ivoz/Provider/Domain/Model/Terminal/TerminalDto.php [hunk 11]
```diff
@@ -2,14 +2,82 @@
...
+     * )
+     */
+    protected $status = [];
+
+    /**
+     * @var string
+     * @AttributeDefinition(
+     *     type="string",
+     *     description="Terminal domain"
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/Terminal/TerminalDto.php]: See above

File: web/rest/client/features/my/call_forward_settings/getCallForwardSettings.feature [hunk 12]
```diff
@@ -1,6 +1,6 @@
 Feature: Retrieve call forward settings
   In order to manage call forward settings
-  As an super admin
+  As a user
```
Comment (Kaian) [inline web/rest/client/features/my/call_forward_settings/getCallForwardSettings.feature]: As a**n** user?

Comment (mmadariaga) [reply web/rest/client/features/my/call_forward_settings/getCallForwardSettings.feature]: Not aparently https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user

Comment (Kaian) [reply web/rest/client/features/my/call_forward_settings/getCallForwardSettings.feature]: Cool, it sound kinda weird tbh 😂

File: library/Ivoz/Provider/Domain/Model/Friend/FriendDto.php (hunk 2 above)
Comment (mmadariaga) [reply library/Ivoz/Provider/Domain/Model/Friend/FriendDto.php]: Thanks :)

File: library/Ivoz/Kam/Domain/Model/UsersLocation/Status.php [hunk 13]
```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Ivoz\Kam\Domain\Model\UsersLocation;
+
+use Ivoz\Api\Core\Annotation\AttributeDefinition;
+
+class Status
```
Comment (Kaian) [inline library/Ivoz/Kam/Domain/Model/UsersLocation/Status.php]: Registration Status could be a good name for this entity

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**